beacon-chain/blockchain/receive_block.go

```diff
@@ -160,11 +166,11 @@ func (s *Service) ReceiveBlock(ctx context.Context, block interfaces.ReadOnlySig
 			log.WithError(err).Error("could not report epoch metrics")
 		}
 	}
-	if err := s.updateJustificationOnBlock(ctx, preState, postState, currStoreJustifiedEpoch); err != nil {
+	if err := s.updateJustificationOnBlock(ctx, preState, postState, cp.j); err != nil {
 		return errors.Wrap(err, "could not update justified checkpoint")
```

Comment:
Unrelated to the PR, I wonder if really should fail `RecieveBlock` if update justification on block fails...

Reply:
Yeah we should fail, having the wrong justification can lead to terribly bad outcomes on both forkchoice and irreversible damage to the database as well.